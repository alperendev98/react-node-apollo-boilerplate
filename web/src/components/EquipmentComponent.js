import React from 'react';
import { getBEMClasses } from '../helpers/cssClassesHelper';
import AdminLayout from '../layouts/AdminLayout';

import '../styles/components/equipmentComponent.css';

const EquipmentClasses = getBEMClasses(['equipment']);

class EquipmentComponent extends React.Component {
	render() {
		return (
			<div className={EquipmentClasses('wrapper')}>
				<div className={EquipmentClasses('container')}>
					<AdminLayout>
						<div>Admin Equipment Component</div>
					</AdminLayout>
				</div>
			</div>
		);
	}
}

export default EquipmentComponent;
