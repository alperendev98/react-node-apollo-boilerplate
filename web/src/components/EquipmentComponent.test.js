import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import EquipmentComponent from './EquipmentComponent';

const mockStore = configureStore();

jest.mock('../layouts/AdminLayout', () => {
	return () => <div className="admin-layout" />;
});

describe('EquipmentComponent', () => {
	const initialState = {};
	let component, wrapper, store;

	beforeEach(() => {
		store = mockStore(initialState);
		component = mount(<EquipmentComponent store={store} />, {});
		wrapper = shallow(<EquipmentComponent />);
	});

	it('shallowly renders EquipmentComponent without crashing', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('EquipmentComponent should render AdminLayout', () => {
		expect(component.find('.admin-layout').length).toBe(1);
	});

	it('should render EquipmentComponent with .equipment__container class', () => {
		expect(component.find('.equipment__container').length).toBe(1);
	});

	it('should match previos snapshot for EquipmentComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
