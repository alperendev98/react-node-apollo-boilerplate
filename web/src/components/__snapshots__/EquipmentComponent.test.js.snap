// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EquipmentComponent should match previos snapshot for EquipmentComponent 1`] = `
<div
  className="equipment__wrapper"
>
  <div
    className="equipment__container"
  >
    <Component>
      <div>
        Admin Equipment Component
      </div>
    </Component>
  </div>
</div>
`;
